vespa/deployment.py

```diff
@@ -307,10 +357,9 @@ def deploy(
         :return: a Vespa connection instance.
         """
         if not self.disk_folder:
-            self.disk_folder = os.path.join(os.getcwd(), application_package.name)
+            return self._execute_deployment_zip(app_package=application_package)
```

Comment:
It would be nice to add an integration test in this PR that uses the functionality implemented here, namely `_execute_deployment_zip` which is triggered when not specifying `disk_folder`.

Reply:
agreed, will do